ActivatePubKey/handler.ts

```diff
@@ -0,0 +1,193 @@
+        AssertionFileName.decode,
+        TE.fromEither,
+        TE.bindTo("assertionFileName"),
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(
+            `Could not decode assertionFileName | ${readableReport(errors)}`
+          )
+        ),
+        TE.chainFirst(({ assertionFileName }) =>
+          pipe(
+            assertionWriter(assertionFileName, body.assertion),
+            TE.mapLeft(error => ResponseErrorInternal(error.detail))
+          )
+        ),
+        TE.bind("jwkPubKeyFromString", () =>
+          pipe(
+            popDocument.pubKey,
+            JwkPublicKeyFromToken.decode,
+            TE.fromEither,
+            TE.mapLeft(errors =>
+              ResponseErrorInternal(
+                `Could not decode public key | ${readableReport(errors)}`
+              )
+            )
+          )
+        ),
+        TE.bind("assertionRefs", ({ jwkPubKeyFromString }) =>
+          pipe(
+            getAllAssertionsRef(
+              JwkPubKeyHashAlgorithmEnum.sha512,
+              getAlgoFromAssertionRef(popDocument.assertionRef),
+              jwkPubKeyFromString
+            ),
+            TE.mapLeft((error: Error) => ResponseErrorInternal(error.message))
+          )
+        ),
+        TE.bind(
+          "retrievedPopDocument",
+          ({ assertionRefs, assertionFileName }) =>
+            pipe(
+              popDocumentWriter({
+                assertionFileName,
+                assertionRef: assertionRefs.master,
+                assertionType: body.assertion_type,
+                expiredAt: body.expires_at,
+                fiscalCode: body.fiscal_code,
+                pubKey: popDocument.pubKey,
+                status: PubKeyStatusEnum.VALID,
+                ttl: TTL_VALUE_AFTER_UPDATE
+              }),
+              TE.mapLeft(error => ResponseErrorInternal(error.detail))
+            )
+        ),
+        TE.bind(
+          "retrievedUsedPopDocument",
+          ({ assertionRefs, assertionFileName }) =>
+            assertionRefs.used
+              ? pipe(
+                  popDocumentWriter({
+                    assertionFileName,
+                    assertionRef: assertionRefs.used,
+                    assertionType: body.assertion_type,
+                    expiredAt: body.expires_at,
+                    fiscalCode: body.fiscal_code,
+                    pubKey: popDocument.pubKey,
+                    status: PubKeyStatusEnum.VALID,
+                    ttl: TTL_VALUE_AFTER_UPDATE
+                  }),
+                  TE.mapLeft(error => ResponseErrorInternal(error.detail))
+                )
+              : TE.of(void 0)
+        ),
+        TE.chain(({ retrievedPopDocument, retrievedUsedPopDocument }) =>
+          pipe(
+            retrievedUsedPopDocument,
+            O.fromNullable,
+            O.getOrElse(() => retrievedPopDocument),
+            TE.fromPredicate(
+              (rd): rd is RetrievedValidPopDocument =>
```

Comment:
move this guard into utils

Reply:
Done in [d450057](https://github.com/pagopa/io-functions-lollipop/pull/13/commits/d4500570e87494922f1c25cc1197a1a58287cc75)